### ISSUE TITLE
Category edit breaks when image is no longer found

### ISSUE THREAD

**Issue by robjens** [NodeBB/NodeBB#4227]
Hi, I first thought it might be the category background I once (a while back) added to some NodeBB test setup categories (since all other uploaded, e.g. profile pics seem missing too now). Yesterday I updated the NodeBB but sadly I hadn't checked categories before that, so I can't really tell if it was ok before or this is through a recent upgrade. For categories, I keep hitting 404's for some reason

![image](https://cloud.githubusercontent.com/assets/11848802/13196640/8f0715d8-d7cb-11e5-8559-55222ac898d4.png)

![image](https://cloud.githubusercontent.com/assets/11848802/13196642/c5f469b0-d7cb-11e5-8092-078397e53ff5.png)

I notice the url has a whole lot of `/categories/categories/categories` nesting, which does not seem to equate to the actual nesting of the category (top level) so it might be just that the URL generation is wack (as it is giving 404). This looks familiar to anyone?

**Comment by barisusakli** [comment]
Can you run in dev mode and check where that `uncaught typeerror` points to.

**Comment by robjens** [comment]
Oh ... how silly of me. Of course I was running it in start only -.-

Ok so the console output is clean, nothing showing that indicates an error server side anyway. I've noticed now though, I **am** capable of clicking through edit and move to the next screen.

Even on those missing the images. I'll see if I can fix those by uploading a new one and inspect the console (browser) output again, currently for the most of it, seems the 404's are all images missing.

First thing (Chrome) complains about the uploader is:
![image](https://cloud.githubusercontent.com/assets/11848802/13223769/93affbb4-d97d-11e5-88a6-98cfba4e3164.png)

Interesting side-effects are:
![image](https://cloud.githubusercontent.com/assets/11848802/13223814/ba7d96a2-d97d-11e5-95c2-86b75b211754.png)
which used to be printed twice, until I uploaded the new category image and now it only prints once. But the weird thing even more so is that there isn't a profile to be seen anywhere around in that part of the admin panels. Only I am logged in (1 user) and I renewed my profile image. There are 4 users on the site (accounts) and I know 1 has a dead image in his profile. The odd thing is of course that it seems to transpile the lack of a (not to be loaded image) to the browser console in general. Also I am user 1 and that is user 2. How does it end up there? What is calling it and why?

Looking into it further though, it seems this is sipping through from the permissions on categories.
![image](https://cloud.githubusercontent.com/assets/11848802/13223913/272016ea-d97e-11e5-8cfc-6bf8b8f9af1b.png)

I guess this is the 'correct' behavior then. As I'm not happy about it and especially not if many users have dead links, I would almost suggest it should (can it?) fallback on an avatar in this case since 404, to begin with, are killing the speed to a far greater degree than any normal (cached/or just 200) succeeding. The problem might be to detect this in the first place, it could just as well add up to the same amount of delay. Anyway, I guess this, when it happens, is your problem anyway since these are images which are expected to be uploaded to the local server machine and thus, if 1 fails it is very likely that a lot more would fail too (I also lost my profile pic, the 2 users who had avatars didn't of course since this aren't uploaded).

Also, I'm still not sure what happend to the uploaded images to begin with but they might be the result of a (destructive) upgrade process cleaning them perhaps?

To continue (I'll just provide some additional feedback as well), I decided to use my ultra-special admin powers to change his profile picture to ensure we have 0 or as few as possible errors in the admin console before continuing to track down this issue.

While (awesome) there is a 'remove image' option in the profile -> edit -> options -> edit image popup, I notice that there is a 'save changes' button. But if I do not click it but remove, it is removed and then close the window and reopen it - it is gone. So this is in fact a destructive operation which doesn't need a 'save' option since it always just saves (or at least removes). I'll check this again when it does have a picture. However, although the list of profile pictures does contain only 1 (avatar) now, the actual profile picture is still a dead link. So it removes it, but doesn't (other record?). I tried F5 and it keeps the dead image pica and doesn't switch to avatar although the (likely) expected behavior would be that it falls back to the only available option, the default avatar. This makes me suspect that there is another database record/field keeping a (dead) url to the picture.

So.. (done) selected the avatar and closed (no save) which has no change (expected). When I open it again and select avatar and save, it does save (expected). So the only problem here seems to be that deleting the profile image always happens regardless of save while you would want/expect it to cancel removing it (bug).

Now ... back in admin panel, I notice the  acp.min is freaking out but see a url parameter that tells me what the problem is. It has `&polling` in it so websocket is failing (indeed, I do not have it setup with redis properly yet).

I'll do that first (it wasn't that trivial last time) and report back but for now, I'm guessing most if not all console messages will be resolved after I've done that. Weird thing still that running normally would stop the whole page navigation while running `./nodebb dev` doesn't.

Thanks for the pointer anyway.
